### ISSUE TITLE
Is there a way to catch setting state?

### ISSUE THREAD

**Issue by moroshko** [yannickcr/eslint-plugin-react#195]
I'd like to check whether I set state in my React app. More specifically, I'd like to catch things like:
```js
this.state = { ... };
maybeSomethingElse.state = { ... };
this.setState({ ... });
maybeSomethingElse.setState({ ... });
```
In other words, I'd like to restrict setting `state` and calling `setState()`.

Is that possible to catch these?

**Comment by moroshko** [comment]
#197 Looks like a good start

**Comment by markdalgleish** [comment]
Ha, didn't realise you opened this issue! Perfect timing :smile:

**Comment by markdalgleish** [comment]
#197 got merged, this issue can be closed now.

**Comment by remitbri** [comment]
Actually, @markdalgleish's `no-set-state` rule only prohibits the use of `setState`. Shouldn't `getInitialState()` be avoided as well in this context, i.e if you forbid one you forbid the other as well? 

From what I understand from @moroshko's initial question, there's still the problem of doing things like `this.state.foo = "bar"`… So should the issue be re-opened? Or maybe a new issue should be created for this particular thing?

**Comment by yannickcr** [comment]
Yest, that the reason why I did not closed this issue at first, since all reported cases are not (yet) handled.
